client/src/routes/settings/+page.svelte

```diff
@@ -21,11 +54,165 @@
+  }
+
+  function passValidator() {
+    if (password == confirmPassword) {
+      linkPassword();
+      clearFields();
+    } else {
+      return;
+    }
+  }
+
+  const googleSvgPaths: { color: string; path: string }[] = [
+    {
+      color: "#EA4335",
+      path: "M24 9.5c3.54 0 6.71 1.22 9.21 3.6l6.85-6.85C35.9 2.38 30.47 0 24 0 14.62 0 6.51 5.38 2.56 13.22l7.98 6.19C12.43 13.72 17.74 9.5 24 9.5z",
+    },
+    {
+      color: "#4285F4",
+      path: "M46.98 24.55c0-1.57-.15-3.09-.38-4.55H24v9.02h12.94c-.58 2.96-2.26 5.48-4.78 7.18l7.73 6c4.51-4.18 7.09-10.36 7.09-17.65z",
+    },
+    {
+      color: "#FBBC05",
+      path: "M10.53 28.59c-.48-1.45-.76-2.99-.76-4.59s.27-3.14.76-4.59l-7.98-6.19C.92 16.46 0 20.12 0 24c0 3.88.92 7.54 2.56 10.78l7.97-6.19z",
+    },
+    {
+      color: "#34A853",
+      path: "M24 48c6.48 0 11.93-2.13 15.89-5.81l-7.73-6c-2.15 1.45-4.92 2.3-8.16 2.3-6.26 0-11.57-4.22-13.47-9.91l-7.98 6.19C6.51 42.62 14.62 48 24 48z",
+    },
+  ];
 </script>
 
 <html lang="en">
-  <h2>Account Settings</h2>
-  <main>
+  <main class="settings-wrapper">
+    <h2>Account Settings</h2>
+    <p id="email">Email: {getEmail()}</p>
+    <div>
+      <Button on:click={() => (showOptions = true)}>
+        <Label>Link Account Options</Label>
+      </Button>
+    </div>
+    <Dialog bind:open={showOptions} aria-labelledby="link-options-title" aria-describedby="link-options-content">
+      <Title id="link-options-title">Link Account Options</Title>
+      <Content id="link-options-content">
+        <div class="signin-buttons">
+          <Button id="sign-in-with-google" variant="raised" on:click={linkGoogleAccount}>
+            <Icon component={Svg} viewBox="0 0 48 48">
+              {#each googleSvgPaths as path}
+                <path fill={path.color} d={path.path} />
+              {/each}
+            </Icon>
+            <Label>Sign in with Google</Label>
+          </Button>
+          {#if googleErr !== ""}
+            <p class="error">{googleErr}</p>
+          {:else if googleLinked && googleErr === ""}
+            <p class="success">Successfully Linked Google Account</p>
+          {/if}
+          <Button id="sign-in-with-microsoft" variant="raised" on:click={linkMicrosoftAccount}>
+            <Icon component={Svg} viewBox="0 0 21 21">
+              <rect x="1" y="1" width="9" height="9" fill="#f25022" />
+              <rect x="1" y="11" width="9" height="9" fill="#00a4ef" />
+              <rect x="11" y="1" width="9" height="9" fill="#7fba00" />
+              <rect x="11" y="11" width="9" height="9" fill="#ffb900" />
+            </Icon>
+            <Label>Sign in with Microsoft</Label>
+          </Button>
+          {#if microsoftErr !== ""}
+            <p class="error">{microsoftErr}</p>
+          {:else if microsoftLinked && microsoftErr === ""}
+            <p class="success">Successfully Linked Microsoft Account</p>
+          {/if}
+        </div>
+        <h3>Set Password</h3>
```

Comment:
works but doesn't closes the dialog after setting the password

Reply:
Closes dialog